File: output/cpp/tests/basic_project.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <any>
#include <memory>
#include <functional>


#include <iostream>
#include <string>
#include <cassert>

class PFrameEvent {
public:
    std::string _message;
    std::unordered_map<std::string, std::any> _parameters;

    PFrameEvent(const std::string& message, std::unordered_map<std::string, std::any> params = {})
        : _message(message), _parameters(std::move(params)) {}
};

class PFrameContext {
public:
    PFrameEvent _event;
    std::any _return;
    std::unordered_map<std::string, std::any> _data;

    PFrameContext(PFrameEvent event, std::any default_return = {})
        : _event(std::move(event)), _return(std::move(default_return)) {}
};

class PCompartment {
public:
    std::string state;
    std::unordered_map<std::string, std::any> state_args;
    std::unordered_map<std::string, std::any> state_vars;
    std::unordered_map<std::string, std::any> enter_args;
    std::unordered_map<std::string, std::any> exit_args;
    std::unique_ptr<PFrameEvent> forward_event;
    std::unique_ptr<PCompartment> parent_compartment;

    explicit PCompartment(const std::string& state) : state(state) {}

    std::unique_ptr<PCompartment> clone() const {
        auto c = std::make_unique<PCompartment>(state);
        c->state_args = state_args;
        c->state_vars = state_vars;
        c->enter_args = enter_args;
        c->exit_args = exit_args;
        return c;
    }
};

class P {
private:
    std::vector<std::unique_ptr<PCompartment>> _state_stack;
    std::unique_ptr<PCompartment> __compartment;
    std::unique_ptr<PCompartment> __next_compartment;
    std::vector<PFrameContext> _context_stack;

    void __kernel(PFrameEvent& __e) {
        __router(__e);
        while (__next_compartment) {
            auto next_compartment = std::move(__next_compartment);
            PFrameEvent exit_event("<$");
            __router(exit_event);
            __compartment = std::move(next_compartment);
            if (!__compartment->forward_event) {
                PFrameEvent enter_event("$>");
                __router(enter_event);
            } else {
                auto forward_event = std::move(__compartment->forward_event);
                if (forward_event->_message == "$>") {
                    __router(*forward_event);
                } else {
                    PFrameEvent enter_event("$>");
                    __router(enter_event);
                    __router(*forward_event);
                }
            }
        }
    }

    void __router(PFrameEvent& __e) {
        const auto& state_name = __compartment->state;
        if (state_name == "A") {
            _state_A(__e);
        } else if (state_name == "B") {
            _state_B(__e);
        }
    }

    void __transition(std::unique_ptr<PCompartment> next) {
        __next_compartment = std::move(next);
    }

    void _state_B(PFrameEvent& __e) {
        if (__e._message == "e") {
            ;
        }
    }

    void _state_A(PFrameEvent& __e) {
        if (__e._message == "e") {
            auto __new_compartment = std::make_unique<PCompartment>("B");
            __new_compartment->parent_compartment = __compartment->clone();
            __transition(std::move(__new_compartment));
            return;
        }
    }

public:
    P() {
        __compartment = std::make_unique<PCompartment>("A");
        PFrameEvent __frame_event("$>");
        PFrameContext __ctx(std::move(__frame_event));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }

    void e() {
        PFrameEvent __e("e");
        PFrameContext __ctx(std::move(__e));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }
};

// Stub functions for placeholder calls
void native() {}
void x() {}

// TAP test harness
int main() {
    printf("TAP version 14\n");
    printf("1..1\n");
    try {
        P s;
        s.e();
        printf("ok 1 - basic_project\n");
    } catch (...) {
        printf("not ok 1 - basic_project\n");
    }
    return 0;
}

File: output/go/tests/basic_project.go

package main

import "fmt"

type PFrameEvent struct {
    _message    string
    _parameters map[string]any
}

type PFrameContext struct {
    _event  PFrameEvent
    _return any
    _data   map[string]any
}

type PCompartment struct {
    state            string
    stateArgs        map[string]any
    stateVars        map[string]any
    enterArgs        map[string]any
    exitArgs         map[string]any
    forwardEvent     *PFrameEvent
    parentCompartment *PCompartment
}

func newPCompartment(state string) *PCompartment {
    return &PCompartment{
        state:    state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
    }
}

func (c *PCompartment) copy() *PCompartment {
    nc := &PCompartment{
        state: c.state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
        forwardEvent:     c.forwardEvent,
        parentCompartment: c.parentCompartment,
    }
    for k, v := range c.stateArgs { nc.stateArgs[k] = v }
    for k, v := range c.stateVars { nc.stateVars[k] = v }
    for k, v := range c.enterArgs { nc.enterArgs[k] = v }
    for k, v := range c.exitArgs { nc.exitArgs[k] = v }
    return nc
}

type P struct {
    _state_stack []*PCompartment
    __compartment *PCompartment
    __next_compartment *PCompartment
    _context_stack []PFrameContext
}

func NewP() *P {
    s := &P{}
    s._state_stack = make([]*PCompartment, 0)
    s._context_stack = make([]PFrameContext, 0)
    s.__compartment = newPCompartment("A")
    s.__next_compartment = nil
    __frame_event := PFrameEvent{_message: "$>", _parameters: nil}
    __ctx := PFrameContext{_event: __frame_event, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
    return s
}

func (s *P) __kernel(__e *PFrameEvent) {
    s.__router(__e)
    for s.__next_compartment != nil {
        next_compartment := s.__next_compartment
        s.__next_compartment = nil
        exit_event := &PFrameEvent{_message: "<$", _parameters: s.__compartment.exitArgs}
        s.__router(exit_event)
        s.__compartment = next_compartment
        if s.__compartment.forwardEvent == nil {
            enter_event := &PFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
            s.__router(enter_event)
        } else {
            forward_event := s.__compartment.forwardEvent
            s.__compartment.forwardEvent = nil
            if forward_event._message == "$>" {
                s.__router(forward_event)
            } else {
                enter_event := &PFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
                s.__router(enter_event)
                s.__router(forward_event)
            }
        }
    }
}

func (s *P) __router(__e *PFrameEvent) {
    switch s.__compartment.state {
    case "A":
        s._state_A(__e)
    case "B":
        s._state_B(__e)
    }
}

func (s *P) __transition(next *PCompartment) {
    s.__next_compartment = next
}

func (s *P) E() {
    __e := PFrameEvent{_message: "E"}
    __ctx := PFrameContext{_event: __e, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
}

func (s *P) _state_B(__e *PFrameEvent) {
    if __e._message == "E" {
        ;
    }
}

func (s *P) _state_A(__e *PFrameEvent) {
    if __e._message == "E" {
        __compartment := newPCompartment("B")
        __compartment.parentCompartment = s.__compartment.copy()
        s.__transition(__compartment)
        return
    }
}

func main() {
	fmt.Println("TAP version 14")
	fmt.Println("1..1")
	s := NewP()
	s.E()
	fmt.Println("ok 1 - basic_project")
}
